_posts/2019-06-06-NeuralODEs.markdown

```diff
@@ -0,0 +1,388 @@
+     </p>
+     </details>
+
+  2. Exercises 12.3,12.4, 12.7 in Section 12 of (Süli & Mayers)
+     <details><summary>Solutions</summary>
+     <p>
+     Exercise 12.3: 
+     Notice that we can write
+     
+     $$(y + q/p)'=p(y + q/p)$$
+     
+     It follows that \(y(t) = Ce^{pt} - q/p\) for some constant \(C\). Imposing the initial condition \(y(0)=1\), we get \(y(t)=e^{pt} + q/p(e^{pt}-1)\). In particular, we expand \(y\) in its Taylor series : \(y(t) = 1 + (1+q/p)\sum_{k=1}^\infty \frac{(pt)^k}{k!}\). To conclude the exercise we only need to notice that \(y_n(t) = q/p + (1+q/p)\sum_{k=1}^n \frac{(pt)^k}{k!}\) satisfies Picard's iteration: \(y_0 \equiv 1\), \(y_{n+1}(t) = y_0 + \int_0^t (py_n(s) + q)\,ds\).
+     </p>
+     <p>
+     Exercise 12.4:
+     Applying Euler's method with step-size \(h\), we get \(\hat{y}(0) = 0\), \(\hat{y}(h) = \hat{y}(0) + h \hat{y}(0)^{1/5} = 0\), \(\hat{y}(2h) = \hat{y}(h) + h \hat{y}(h)^{1/5} =0\). Iterating, we see that it holds \(y(nh)=0\) for all \(n\geq 0\). Instead, the implicit Euler's method reads
+     
+     $$\hat{y}_{n+1} = \hat{y}_n + h \hat{y}_{n+1}^{1/5}$$
+     
+     for \(n \geq 0\) and \(\hat{y}_0 = 0\). After substituting \(\hat{y}_{n} = (C_nh)^{5/4}\) in the above relation, we only need that there exists a sequence \(C_n\) satisying the requirements.
+     </p>
+     <p>
+     Exercise 12.7: 
+     First, notice that
+     
+     $$e_{n+1} = y(x_{n+1}) - y_{n} - \frac{1}{2}h(f_{n+1} + f_n)= e_n - \frac{1}{2}h (f_{n+1}+f_n) + \int_{x_n}^{x_{n+1}} f(s,y(s))\,ds$$
+     
+     and that the second component of the RHS is the same as \(E_1\) in Exercise 1 above. Therefore the first bound follows. The last inequality is simply obtained by re-arranging the terms.  
+     </p>
+     </details>
+
+  3. Consider the following method for solving \(y' = f(y)\):
+       
+     $$y_{n+1} = y_n + h*(theta*f(y_n) + (1-theta)*f(y_{n+1}))$$
+    
+     Assuming sufficient smoothness of \(y\) and \(f\), for what value of \(0 \leq\theta\leq 1\) is the truncation error the smallest? What does this mean about the accuracy of the method?
+     <details><summary>Solution</summary>
+     <p>
+     By definition, it holds
+     
+     $$h\tau_n = y_{n+1} - y_n - h (\theta f_n + (1-\theta) f_{n+1}) = y_{n+1} - y_n - h \theta y_n' - h(1-\theta) y_{n+1}'$$
+     
+     Taylor-expanding, we get
+     
+     $$h\tau_n = y_{n} + hy_n' + h^2/2y_n'' + O(h^3) - y_n - h \theta y_n' - h(1-\theta) y_{n}' - h^2(1-\theta) y_{n}'' + O(h^3) = h^2(\theta - 1/2)y_n''+O(h^3)$$
+     
+     It follows that the truncation error is the smallest for \(\theta=1/2\). For \(\theta = 1/2\) the method has order \(2\), otherwise it has order \(1\).
+     </p>
+     </details>
+
+  4. [Colab Notebook](https://colab.research.google.com/drive/1bNg-RzZoelB3w8AUQ6mefRQuN3AdrIqX)
+     <details><summary>Solution</summary>
+     <p>
+     [Solution](https://colab.research.google.com/drive/1wTQXy2_4InQH51rEmiCtvl5Q7MiyrC4k)
```

Comment:
Solution 4:
	- The link is not working.
	- It should also prolly say “Colab” or "See this Colab for the solution." instead of "Solution".

Reply:
fixed

Reply:
the link is working to me, does it work to you now?

Reply:
when viewing it, this is what it shows:
'See this [Colab](https://colab.research.google.com/drive/1wTQXy2_4InQH51rEmiCtvl5Q7MiyrC4k) for the solution.'

Reply:
fixed